lib/modules/decoder/PayloadService.ts

```diff
@@ -174,6 +174,20 @@ export class PayloadService extends BaseService {
         ),
       );
 
+    // Due to the existence of a "devices" collection in the tenant index and a platform index,
+    // it is need to fetch the device content associated to the tenant if it exists.
+    const updatedDevices = await Promise.all(
+      devices.map((device) =>
+        device._source.engineId && device._source.engineId.trim() !== ""
+          ? this.sdk.document.get<DeviceContent>(
```

Comment:
Doing ```get request``` in a ```map``` does not seems good but if there is a better solution, I'll take it.